force-app/main/default/classes/ReceiptTriggerHandler.cls

```diff
@@ -0,0 +1,106 @@
+// 3.    When Amount of Receipt object is changed
+// 3.1    Check sum of “Amount of payments” of all Cases related to current doctor in current month– if received sum is more than “Doctor payments to receive bonus” field value from Hospital department – update field “Payments amount” on Doctor object with received value
+// 3.2    Check the doctors related to the same department with populated “Payments amount” field and set the flag “Best doctor of the month” on object with the higher value. NOTE: There can be only one Best doctor for each hospital department.
+public with sharing class ReceiptTriggerHandler extends TriggerHandler{
+    protected override void afterInsert(){
+        updateTreatmentPaymentsAmount((Map<Id, Receipt__c>)Trigger.newMap);
+        updateDoctorsPaymentsAmount((Map<Id, Receipt__c>)Trigger.newMap);
+    }
+
+    private static void updateTreatmentPaymentsAmount(Map<Id, Receipt__c> newReceiptsMap){
+        List<Case> treatmentsList = [SELECT Id, (SELECT Id
+                                                 FROM Receipts__r
+                                                 WHERE Id IN:newReceiptsMap.keySet())
+                                     FROM Case ];
+        List<Id> treatmentIdsList = new List<Id>();
+        for (Case treatment_i : treatmentsList){
+            treatmentIdsList.add(treatment_i.Id);
+        }
+
+        List<AggregateResult> treatmentReceiptsList = [SELECT SUM(Amount__c)Amount, Treatment__c TrId
+                                                       FROM Receipt__c 
+                                                       WHERE Treatment__c IN:treatmentIdsList
+                                                       GROUP BY Treatment__c];
+
+        List<Case> treatmentsToUpdateList = new List<Case>();
+        for (AggregateResult treatment_i : treatmentReceiptsList){
+            treatmentsToUpdateList.add(new Case(Id = (Id) treatment_i.get('TrId'), Amount_of_payments__c = (Decimal) treatment_i.get('Amount')));
+        }
+        List<Database.SaveResult> updateResults = Database.update (treatmentsToUpdateList, true);
+    }
+
+    private static void updateDoctorsPaymentsAmount(Map<Id, Receipt__c> newReceiptsMap){
+
+        List<Id> listCasesIdsFromReceipts = new List<Id>();
+        for (Receipt__c receipt_i : newReceiptsMap.values()){
+            {
+                listCasesIdsFromReceipts.add(receipt_i.Treatment__c);
+            }
+        }
+        List<Contact> doctorsList = [SELECT Id, AccountId, Account.Department__c, Payments_amount__c, Account.Doctor_payments_to_receive_bonus__c, (SELECT Id, Amount_of_payments__c
```

Comment:
In this case you query all Contacts exist in the system, it can cause limit exception. This SOQL can be written in another way